test/RoadRegistry.Tests/BackOffice/Api/ZipArchiveWriters/ForEditor/GradeSeparatedJunctionArchiveWriterTests.cs

```diff
@@ -0,0 +1,79 @@
+﻿namespace RoadRegistry.BackOffice.Api.ZipArchiveWriters.ForEditor
+{
+    using System;
+    using System.IO;
+    using System.IO.Compression;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Be.Vlaanderen.Basisregisters.Shaperon;
+    using Editor.Schema;
+    using Editor.Schema.GradeSeparatedJunctions;
+    using RoadRegistry.Framework.Containers;
+    using Xunit;
+
+    [Collection(nameof(SqlServerCollection))]
+    public class GradeSeparatedJunctionArchiveWriterTests
+    {
+        private readonly SqlServer _fixture;
+
+        public GradeSeparatedJunctionArchiveWriterTests(SqlServer fixture)
+        {
+            _fixture = fixture ?? throw new ArgumentNullException(nameof(fixture));
+        }
+
+        [Fact]
+        public Task ArchiveCanNotBeNull()
+        {
+            var sut = new GradeSeparatedJunctionArchiveWriter(_fixture.MemoryStreamManager, Encoding.UTF8);
+            return Assert.ThrowsAsync<ArgumentNullException>(
+                () => sut.WriteAsync(null, new EditorContext(), default));
+        }
+
+        [Fact]
+        public Task ContextCanNotBeNull()
+        {
+            var sut = new GradeSeparatedJunctionArchiveWriter(_fixture.MemoryStreamManager, Encoding.UTF8);
+            return Assert.ThrowsAsync<ArgumentNullException>(
+                () => sut.WriteAsync(new ZipArchive(Stream.Null, ZipArchiveMode.Create, true), null, default));
+        }
+
+        [Fact]
+        public async Task WithEmptyDatabaseWritesArchiveWithExpectedEntries()
+        {
+            var sut = new GradeSeparatedJunctionArchiveWriter(_fixture.MemoryStreamManager, Encoding.UTF8);
+
+            var db = await _fixture.CreateDatabaseAsync();
+            var context = await _fixture.CreateEditorContextAsync(db);
+
+            await new ZipArchiveScenario<EditorContext>(_fixture, sut)
+                .WithContext(context)
+                .Assert(readArchive =>
+                {
+                    Assert.Single(readArchive.Entries);
+                    foreach (var entry in readArchive.Entries)
+                    {
+                        switch (entry.Name)
+                        {
+                            case "RltOgkruising.dbf":
+                                using (var entryStream = entry.Open())
+                                using (var reader = new BinaryReader(entryStream, Encoding.UTF8))
+                                {
+                                    Assert.Equal(
+                                        new DbaseFileHeader(
+                                            DateTime.Now,
+                                            DbaseCodePage.Western_European_ANSI,
+                                            new DbaseRecordCount(0),
+                                            GradeSeparatedJunctionDbaseRecord.Schema),
+                                        DbaseFileHeader.Read(reader));
+                                }
```

Comment:
I'd assert that the expected number of records are written ... you could generate a random yet bounded number of database records (e.g. [0,3]) and assert that those records are in the dbase file. It's a bit more complex but it covers more of what your SUT is actually doing (same for all other writers). You could lean on AutoFixture to generate the proper records IF you teach it how to compose such a record (you have all the value objects at your disposal which can translate themselves back to the dbase record primitives).

Reply:
> I'd assert that the expected number of records are written

Do you mean like this?
https://github.com/Informatievlaanderen/road-registry/pull/26/files/eec380baa3b1fd363f0b6a08e8784bf5e591d317#diff-12cfcf5efa164497a681ef5c4321f735R52

The goal for this PR was to have at least have minimal test setups where they were lacking for zip archive writers, since we found the issue with running the archive writers on an empty db. 

I agree on the tests you suggest, can we do that in another PR?

Reply:
You can, but I'd not postpone it too long, given the investment done so far (this would be the last mile).